src/dsst_defacing_wf.py

```diff
@@ -58,153 +60,113 @@ def get_sess_dirs(subj_dir_path, mapping_dict):
-        for dirpath in anat_dir.glob('*'):
-            if dirpath.name.startswith('workdir') or dirpath.name.endswith('QC'):
-                shutil.move(dirpath, intermediate_files_dir)
-
-        # if not no_clean:
-        #     shutil.rmtree(intermediate_files_dir)
-
-
-def generate_3d_renders(defaced_img, render_outdir):
-    rotations = [(45, 5, 10), (-45, 5, 10)]
-    for idx, rot in enumerate(rotations):
-        yaw, pitch, roll = rot[0], rot[1], rot[2]
-        outfile = render_outdir.joinpath('defaced_render_' + str(idx) + '.png')
-        fsleyes_render_cmd = f"fsleyes render --scene 3d -rot {yaw} {pitch} {roll} --outfile {outfile} {defaced_img} -dr 20 250 -in spline -bf 0.3 -r 100 -ns 500"
-        print(fsleyes_render_cmd)
-        run_command(fsleyes_render_cmd)
-
-
-def create_defacing_id_list(qc_dir):
-    rel_paths_to_orig = [re.sub('/orig.nii.gz', '', str(o.relative_to(qc_dir))) for o in qc_dir.rglob('orig.nii.gz')]
-    with open(qc_dir / 'defacing_id_list.txt', 'w') as f:
-        f.write('\n'.join(rel_paths_to_orig))
-
-
-def vqcdeface_prep(input_dir, defacing_output_dir):
-    defacing_qc_dir = defacing_output_dir.parent / 'QC_prep' / 'defacing_QC'
-    interested_files = [f for f in defacing_output_dir.rglob('*.nii.gz') if
-                        'workdir' not in str(f).split('/')]
-    for defaced_img in interested_files:
-        entities = defaced_img.name.split('.')[0].split('_')
-        vqcd_subj_dir = defacing_qc_dir / f"{'/'.join(entities)}"
-        vqcd_subj_dir.mkdir(parents=True, exist_ok=True)
-
-        defaced_link = vqcd_subj_dir / 'defaced.nii.gz'
-        if not defaced_link.exists():
-            defaced_link.symlink_to(defaced_img)
-            generate_3d_renders(defaced_img, vqcd_subj_dir)
-
-        img = list(input_dir.rglob(defaced_img.name))[0]
-        img_link = vqcd_subj_dir / 'orig.nii.gz'
-        if not img_link.exists(): img_link.symlink_to(img)
+def main():
+    # get command line arguments
+    args = get_args()
 
-    create_defacing_id_list(defacing_qc_dir)
+    input_dir = Path(args.bids_dir).resolve()
+    output = Path(args.output_dir).resolve()
+    no_clean = args.no_clean
 
-    vqcdeface_cmd = f"vqcdeface -u {defacing_qc_dir} -i {defacing_qc_dir / 'defacing_id_list.txt'} -m orig.nii.gz -d defaced.nii.gz -r defaced_render"
+    to_deface = []
+    # only for one subject (and all their sessions, if present)
+    if args.participant_label:
+        to_deface = glob(os.path.join(args.bids_dir, args.participant_label, "ses-*"))
+        if to_deface == []:
+            to_deface = glob(os.path.join(args.bids_dir, args.participant_label))
 
-    return vqcdeface_cmd
+    # only for one subset of sessions
+    if args.session_id:
+        to_deface = glob(os.path.join(args.bids_dir, "sub-*", args.session_id))
 
+    # for all sessions
+    session_check = glob(os.path.join(args.bids_dir, "sub-*", "ses-*"))
+    if session_check != []:
+        to_deface = session_check
```

Comment:
If you accept the above suggestion, then you'll have to delete these lines too. I couldn't highlight all of lines 72-90 coz there are some deleted lines of code in that same range of lines (which is not supported by GitHub apparently). 
```suggestion
```